src/test/components/MenuButtons.test.js

```diff
@@ -14,56 +14,210 @@ import {
+    const getInnerShareButton = () => getByValue('Share');
+    const getShareWithUrlsButton = () => getByValue('Share with URLs');
+    const getInnerShareWithUrlsButton = (): null | HTMLElement => {
+      const shareWithUrlsButton = getShareWithUrlsButton();
+      if (
+        shareWithUrlsButton.parentElement !== undefined &&
+        shareWithUrlsButton.parentElement !== null &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== undefined &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== null
+      ) {
+        return shareWithUrlsButton.parentElement.nextElementSibling.getElementsByClassName(
+          'arrowPanelOkButton'
+        )[0];
+      }
+      return null;
+    };
+
+    const getMenuButtonsContainer = () =>
+      getByTestId('menuButtonsCompositeButtonContainer');
+    const getPermalinkButton = () => getByValue('Permalink');
+
     return {
       store,
-      container,
+      ...renderResult,
+      getShareButton,
+      getInnerShareButton,
+      getShareWithUrlsButton,
+      getInnerShareWithUrlsButton,
+      getMenuButtonsContainer,
+      getPermalinkButton,
     };
   }
 
-  // profile.meta.networkURLsRemoved flag is set to false as a default.
-  it('renders the MenuButtons buttons', () => {
-    const { store, container } = setup();
-    store.dispatch(startSymbolicating());
+  describe('share and save buttons', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
 
-    // MenuButtons is rendering a fragment with several children. We need to
-    // check all children to assess that the component renders properly.
-    expect(Array.from(container.children)).toMatchSnapshot();
+    beforeAll(function() {
+      (window: any).TextEncoder = TextEncoder;
+    });
 
-    store.dispatch(doneSymbolicating());
-    expect(Array.from(container.children)).toMatchSnapshot();
-  });
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
 
-  it('renders the MenuButtons buttons with profile.meta.networkURLsRemoved set to true', () => {
-    const profile = processProfile(createGeckoProfileWithJsTimings());
-    profile.meta.networkURLsRemoved = true;
-    const { store, container } = setup(profile);
-    store.dispatch(startSymbolicating());
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
 
-    expect(Array.from(container.children)).toMatchSnapshot();
+    // profile.meta.networkURLsRemoved flag is set to false as a default.
```

Comment:
Is this comment in the right place? I'm not sure why it's relevant here. Does this need to be encoded in an assertion in a test?

Reply:
I added this comment because other 2 tests below were testing other possible values of this flag. So I wanted to explicitly say it that this value is also tested here. But I removed the other tests(see the reason below) so there is no need to explicitly state it.